numpy/core/src/multiarray/methods.c

```diff
@@ -1026,13 +1026,16 @@ any_array_ufunc_overrides(PyObject *args, PyObject *kwds)
     /* check outputs, if any */
     nout = PyUFuncOverride_GetOutObjects(kwds, &out_kwd_obj, &out_objs);
     if (nout < 0) {
+        Py_XDECREF(out_kwd_obj);
         return -1;
     }
     for (i = 0; i < nout; i++) {
         if (PyUFunc_HasOverride(out_objs[i])) {
+            Py_XDECREF(out_kwd_obj);
```

Comment:
Can never be null here, so plain DECREF is fine

Reply:
It seems simpler to me to return NULL in out_kwd_obj. Is there a style guide ref for things like that?

Reply:
I have no reference to give here, but I've might exist. One argument I can give in favor is that it allows you to eliminate the branching that happens inside XDECREF.

I might need to remind myself tomorrow about why this argument exists in the first place though.

Reply:
> why this argument exists in the first place

We need it to hang on to a reference to the result of `PySequence_Fast`.